### PR TITLE
Handle unhandled exceptions/promise rejects in VxSuite servers

### PR DESCRIPTION
**Pull request by kofi-q**
## Overview

Resolves https://github.com/votingworks/vxsuite/issues/4754

Adding unhandled exception and promise rejection handlers to our web servers.
Logs the errors and then shuts down, to avoid continuing in an undefined state.

### API Servers:
```
[backend:run] {"source":"vx-mark-scan-backend","eventId":"application-startup","eventType":"application-status","user":"system","message":"VxMarkScan backend running at http://localhost:3002/","disposition":"success"}
[mark-scan:build]
[mark-scan:build] 3:21:17 PM - Found 0 errors. Watching for file changes.
[backend:run] {"source":"vx-mark-scan-backend","eventId":"unknown-error","eventType":"application-status","user":"system","message":"server shutting down due to unhandledRejection: <famous last words> | Error: <famous last words>\n    at Timeout._onTimeout (/home/vx/dev/src/github.com/votingworks/vxsuite/apps/mark-scan/backend/build/server.js:86:24)\n    at listOnTimeout (node:internal/timers:559:17)\n    at processTimers (node:internal/timers:502:7)","disposition":"na"}
[backend:run]  ELIFECYCLE  Command failed with exit code 1.
[backend:run] [nodemon] app crashed
```
### Asset Servers/Proxies:
```
❯ make run
cd prodserver && node index.js
[HPM] Proxy created: /api  -> http://localhost:3002/
[HPM] Proxy created: /dock  -> http://localhost:3002/
[HPM] Proxy created: /card  -> http://localhost:3001/
{"source":"vx-mark-scan-frontend-server","eventId":"application-startup","eventType":"application-status","user":"system","message":"Ballot Marking Device running at http://localhost:3000/","disposition":"success"}
{"source":"vx-mark-scan-frontend-server","eventId":"unknown-error","eventType":"application-status","user":"system","message":"server shutting down due to unhandledRejection: <famous last words> | Error: <famous last words>\n    at Timeout._onTimeout (/home/vx/dev/src/github.com/votingworks/vxsuite/apps/mark-scan/frontend/prodserver/index.js:24:12)\n    at listOnTimeout (node:internal/timers:559:17)\n    at processTimers (node:internal/timers:502:7)","disposition":"na"}
make: *** [Makefile:28: run] Error 1
```

## Testing Plan
- Manual for now - tough to test via jest, but could potentially add later if necessary.

## Checklist

- [x] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [x] I have added a screenshot and/or video to this PR to demo the change
- ~[ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates~

### CODE REVIEW

File: libs/logging/src/base_types/log_source.ts [hunk 1]
```diff
@@ -1,14 +1,19 @@
 export enum LogSource {
   System = 'system',
   VxAdminFrontend = 'vx-admin-frontend',
+  VxAdminFrontendServer = 'vx-admin-frontend-server',
```
Comment (carolinemodic) [inline libs/logging/src/base_types/log_source.ts:4]: A little pedantic to make this a new set of names but I suppose its more accurate

**Review by carolinemodic — APPROVED** [review]
love it

File: libs/logging/src/base_types/log_source.ts (hunk 1 above)
Comment (kofi-q) [reply libs/logging/src/base_types/log_source.ts:4]: Yeah, noticed that the `prodserver` logs weren't getting logged - we have logic to skip logging for the frontend source IDs if there's no kiosk browser - so, considered switching to using the backend source IDs, but thought that could be confusing in the logs

**Comment by arsalansufi** [comment]
Woohoo this will be so useful for cert QA!

**Pull request merged**